runner-core/src/main/java/org/apache/apisix/plugin/runner/handler/RpcCallHandler.java

```diff
@@ -229,7 +212,7 @@ private void handleExtraInfo(ChannelHandlerContext ctx, ExtraInfoResponse reques
             }
         } else if (EXTRA_INFO_RESP_BODY_KEY.equals(varsKey)) {
             if (!Objects.isNull(postReq)) {
-                postReq.setBody(result);
+                postReq.setBody( new String(result));
```

Comment:
A PR is best to do only one thing, this seems to be no need to modify

Reply:
reverted